numba/np/linalg.py

```diff
@@ -1579,8 +1579,34 @@ def twoD_impl(b, bcpy, n):
         return twoD_impl
 
 
+def _lstsq_resolve_rcond(rcond, dtype, m, n):
+    pass
+
+
+@overload(_lstsq_resolve_rcond, prefer_literal=True)
+def ol_lstsq_resolve_rcond(rcond, dtype, m, n):
+    if isinstance(rcond, types.StringLiteral):
+        if rcond.literal_value == "warn": # it's the default
+            def impl(rcond, dtype, m, n):
+                return np.float64(-1.0)
+            return impl
```

Comment:
There are no tests for this.
And, should it raise a warning?

Reply:
Thanks, I've now written a test. I'm not sure about Numba re-implementing NumPy's warnings, the same question occurs for NumPy 1.22 in `np.MachAr` which is deprecated. I think the conclusion from an OOB conversation was to replicate the warnings from the typing layer such that users are warned about changes to NumPy behaviour/API even though the NumPy implementation itself is not in use. I'll have a go at implementing that here.

Reply:
Implemented in bfad183